vector/src/main/java/im/vector/app/features/home/HomeActivity.kt

```diff
@@ -520,7 +521,16 @@ class HomeActivity :
         if (views.drawerLayout.isDrawerOpen(GravityCompat.START)) {
             views.drawerLayout.closeDrawer(GravityCompat.START)
         } else {
-            super.onBackPressed()
+            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.R && supportFragmentManager.backStackEntryCount == 0) {
```

Comment:
looks like we could extract this logic to reduce the duplication, either extension or base class function 

something like 

```kotlin
if (views.drawerLayout.isDrawerOpen(GravityCompat.START)) { 
  views.drawerLayout.closeDrawer(GravityCompat.START)
} else {
  validateBackPressed { super.onBackPressed() }
}

...

fun Activity.validateBackPressed(onBackPressed: () -> Unit) {
    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.R && supportFragmentManager.backStackEntryCount == 0) {
        if (isTaskRoot) {
            onBackPressed()
        } else {
            Timber.e("Application is potentially corrupted by an unknown activity")
            finishAffinity()
        }
    } else {
        onBackPressed()
    }
}

```

Reply:
Done